error.go

```diff
@@ -36,29 +38,39 @@ func MakeErrorFromHTTP(resp *http.Response) error {
 
 	var errCode int
 	switch resp.StatusCode {
+	case 400:
+		errCode = ErrorInvalidArgument
 	case 403:
 		errCode = ErrorNotAuthorized
 	case 404:
 		errCode = ErrorNotFound
-	case 400:
-		errCode = ErrorInvalidArgument
+	case 422:
```

Comment:
Yes, agreed - 409 is a better fit. I've used 422 in the past for more complex errors where I had to return more context as headers, but if we're just communicating a name collision, 409 is better.